### ISSUE TITLE
team up?

### ISSUE THREAD

**Issue by ericcarraway** [ephetic/CodingBatJS#1]
Hi Brian -
I started on a JS port of the codingbat problems a couple of months ago.  I'm surprised no one else has done it.  What I have going so far uses QUnit and I've started to enter the test conditions for each problem.  Are you interested in working together?
-Eric

**Comment by ephetic** [comment]
Yeah, I'm totally down for that.  I just started stubbing things today, so working on your project is probably better.  Which repo is it?

**Comment by ericcarraway** [comment]
I'll have to put it up.  Let me see if I can find an example really quick and make a repo.

**Comment by ericcarraway** [comment]
I mocked out a rough directory structure and an idea for seeing tests in a browser.  Here's an example using one of my favorite problems https://github.com/ericcarraway/codingbat-js/tree/master/src/logic-2/makeChocolate  
My idea is that someone would clone or fork the repo and enter their solution in the `solution` folder, while being able to see a possible solution in the `solution-completed` folder.  
I don't want to use qUnit, but I have it here because it was quick and easy to set up.  I'd really like to use Mocha, Chai, and Sinon if you think that's the best option (I saw you started with that).

**Comment by ephetic** [comment]
I'm doing a bootcamp right now and they use mocha/chai/sinon for all the
front end tests, so yeah that's what I was thinking -- the people who
benefit from these probably aren't going to be comfortable forking a repo.
I was thinking one page per CodingBat section (e.g. Recursion-1), but your
README.md does make a nice problem page.

**Comment by ericcarraway** [comment]
Yeah, ultimately a "code in the browser" setup would be great.  Maybe running live at a web address that wouldn't require any setup for a casual user.

I'd like to learn to use mocha/chai/sinon, but I've got no experience with it.

Which bootcamp?  I attended CoderCamps Houston last summer.

**Comment by ephetic** [comment]
MakerSqure in Los Angeles.  Just finished second week.  Feels like a month
so far.

I overlooked the IDE and sandbox parts of code in browser, so I think
starting with a fork or download strategy makes sense after all, to start
with at least.

As for qUnit (looking at it now) vs the others, we haven't done our testing
sprint yet, so I can't speak to the advantages of one of the other.

**Comment by ericcarraway** [comment]
Sounds like a bootcamp!  It was this week a year ago that I hit rock bottom.  I was in week 6 of 9.  Be prepared for huge fluctuations in confidence.

I think QUnit is the way to go for now, since it's easy and can run alone in a browser without dependencies.

With a very clear readme file, cloning, forking, or downloading a zip file will definitely be easiest.  I've got an idea for a sandbox that can run live on a website, but I want to get this running first.

**Comment by ephetic** [comment]
I wouldn't say I've hit bottom yet, but I am learning new things about my
body's response to stress. I'm setting a reminder on your email for another
week out.

Thanks
Brian

**Comment by ericcarraway** [comment]
:+1:

**Comment by ephetic** [comment]
Closer to bottom.  Bumping you another week.  :\

**Comment by ericcarraway** [comment]
:+1: :+1:

**Comment by ephetic** [comment]
Did you just drop a pic on my pigly site?  It was right as I was
presenting, which was hilarious.

**Comment by ericcarraway** [comment]
Nope, but I did almost drop a "hang in there" cat GIF here as a response to your message from last week.  Just skimmed over the piggly repo.  Hadn't seen EJS templating before.  Looks cool.  How's camp?

**Comment by ephetic** [comment]
One of the staff who was watching the presentation finally copped to it.
It was pretty funny.

So be aware that that code is fugly because it was the Minimum Viable
Product sprint -- think of and implement something solo in a little over a
day.  Since you mentioned the EJS, you probably just looked at the master
branch.  The bulk of it is on the angular branch.  Master was the first
draft with a non-ajax API.  The angular version doesn't use the EJS at
all.  Wah wah.

b

**Comment by ericcarraway** [comment]
How's MakerSquare going?

I looked around for awhile for a viable javascript sandbox.  Ideally something that evaluates code and runs tests in a browser without requiring a user to pull a repo down and run it locally.  Only a few good things are out there, most notably this https://github.com/codeschool/abecedary

...Then I thought about JSFiddle, CodePen, and how I might be trying to reinvent the wheel.  I put this up on Plunker.  http://plnkr.co/edit/JnkDlo?p=preview

A good compromise would be to have a GitHub repo with all of the files one would need to run the problems locally, but also include links to Plunker where an environment would already be set up with tests.

![screen shot 2015-08-28 at 11 00 06 pm](https://cloud.githubusercontent.com/assets/6632454/9560464/ddf2a228-4dda-11e5-891a-c09346cdab1a.png)

**Comment by ephetic** [comment]
That looks pretty good.

So I just started my "thesis" phase at MakerSquare, so I'm still getting
totally crushed, but I am going to come back to this.  It is on my short
list of during-job-search projects.
b

**Comment by ericcarraway** [comment]
How did MakerSquare go?

**Comment by ephetic** [comment]
Intense.  Pretty good overall.  The first six weeks were learning to
program (for those who didn't already know) and learning popular libraries
and frameworks, then the second 6/7 weeks were projects and (a little job
search and resume type stuff).  I was not super happy with the project
phase because they had us generate the projects AND manage the teams
ourselves.  We hadn't done anything on PMing or organization -- or even
architecture really, so I thought it was a big missed opportunity for
intensional learning.

Their career prep could be summed up as: Maneuver past the HR filters.
Apply to tons of jobs.

I got a good job after two months of obstinate looking -- but not as shiny
and cool as they were pushing for (not in Santa Monica, not at a start up,
not getting paid in gold and strippers).  I think it's insane they're
pushing students to go after mid-level jobs with just a bootcamp under
their belt, but I'm super supportive of (most) bootcamp grads being ready
for junior level jobs.  I think a bootcamp is a probably a substitute for a
non-elite CS BS -- software is definitely more of a trade than a science.

Post bootcamp: I feel like I'm finally seeing the bigger picture at my new
job.  I see now why novel programming languages are so slow to take off: in
the big picture, all languages are largely equivalent and programming is
just a task, so professionals are going to spend their energy elsewhere
(until evangelists have made the advantages look substantial enough).

That turned out a little long...

Brian
